src/main/java/com/celements/job/AbstractJob.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.celements.job;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.MDC;
+import org.xwiki.context.Execution;
+import org.xwiki.context.ExecutionContext;
+import org.xwiki.context.ExecutionContextException;
+import org.xwiki.context.ExecutionContextManager;
+
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.plugin.scheduler.XWikiServletRequestStub;
+import com.xpn.xwiki.plugin.scheduler.XWikiServletResponseStub;
+import com.xpn.xwiki.web.Utils;
+import com.xpn.xwiki.web.XWikiResponse;
+import com.xpn.xwiki.web.XWikiServletRequest;
+
+/**
+ * Base class for any XWiki Quartz Job. This class take care of initializing ExecutionContext
+ * properly.
+ * <p>
+ * A class extending {@link AbstractJob} should implements {@link #executeJob(JobExecutionContext)}.
+ *
+ * @since 2.90
+ * @author fabian pichler
+ */
+public abstract class AbstractJob implements Job {
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.quartz.Job#execute(org.quartz.JobExecutionContext)
+   */
+  @Override
+  public final void execute(JobExecutionContext jobContext) throws JobExecutionException {
+    JobDataMap data = jobContext.getJobDetail().getJobDataMap();
+
+    // The XWiki context was saved in the Job execution data map. Get it as we'll retrieve
+    // the script to execute from it.
+    XWikiContext xwikiContext = (XWikiContext) data.get("context");
+
+    Execution execution;
+    try {
+      execution = initExecutionContext(xwikiContext);
+    } catch (ExecutionContextException | DocumentNotExistsException | MalformedURLException exp) {
+      throw new JobExecutionException("Fail to initialize execution context", exp);
+    }
+
+    try {
+      // Execute the job
+      executeJob(jobContext);
+    } catch (Throwable t) {
+      getLogger().error("Exception thrown during job execution", t);
```

Comment:
`throw t` again, otherwise you break caller exception handling in `org.quartz.core.JobRunShell.run()`
```
try {
  job.execute(jec);
} catch (JobExecutionException jee) {
  ...
} catch (Throwable e) {
  ...
}
```